src/cleanlab_codex/response_validation.py

```diff
@@ -253,8 +329,35 @@ def is_unhelpful_response(
                                        E.g. if confidence_score_threshold is 0.5, then responses with scores higher than 0.5 are considered unhelpful.
 
     Returns:
-        bool: `True` if TLM determines the response is unhelpful with sufficient confidence,
-              `False` otherwise.
+        SingleResponseValidationResult: A SingleResponseValidationResult object containing the results of the validation checks.
+    """
+    score: float = score_unhelpful_response(response, query, tlm)
+
+    # Current implementation assumes question is phrased to expect "Yes" for unhelpful responses
+    # Changing the question would require restructuring this logic and potentially adjusting
+    # the threshold value in BadResponseDetectionConfig
```

Comment:
Now that the code for scoring has been moved to a separate function, this comment is a bit hard to understand. I think the following would be clearer:
```suggestion
    # Current implementation of `score_unhelpful_response` produces a score where a higher value means the response if more likely to be unhelpful
    # Changing the TLM prompt used in `score_unhelpful_response` may require restructuring the logic for `fails_check` and potentially adjusting
    # the threshold value in BadResponseDetectionConfig
```